db/migrate/20180213163900_create_achievement_users.rb

```diff
@@ -0,0 +1,9 @@
+class CreateAchievementUsers < ActiveRecord::Migration[5.0]
+  def change
+    create_table :achievement_users do |t|
+      t.references :achievement, foreign_key: true, index: true, unique: true
```

Comment:
`unique: true`will only allow a achievement to be listed in this table once. The effect of this is that only one user can receive an achievement